plugins/chat/app/models/chat/thread.rb

```diff
@@ -126,17 +126,20 @@ def self.update_counts
       # It is updated eventually via Jobs::Chat::PeriodicalUpdates. In
       # future we may want to update this more frequently.
       updated_thread_ids = DB.query_single <<~SQL
-        UPDATE chat_threads threads
-        SET replies_count = subquery.replies_count
-        FROM (
-          SELECT COUNT(*) - 1 AS replies_count, thread_id
-          FROM chat_messages
-          WHERE chat_messages.deleted_at IS NULL AND thread_id IS NOT NULL
-          GROUP BY thread_id
-        ) subquery
-        WHERE threads.id = subquery.thread_id
-        AND subquery.replies_count != threads.replies_count
-        RETURNING threads.id AS thread_id;
+        WITH updated_threads AS (
+          UPDATE chat_threads ct
+          SET replies_count = COALESCE(subquery.new_count, 0)
+          FROM (
+            SELECT cm.thread_id, COUNT(cm.*) - 1 AS new_count
+            FROM chat_threads
+            LEFT JOIN chat_messages cm ON cm.thread_id = chat_threads.id AND cm.deleted_at IS NULL
+            GROUP BY cm.thread_id
+          ) AS subquery
+          WHERE ct.id = subquery.thread_id AND ct.replies_count IS DISTINCT FROM COALESCE(subquery.new_count, 0)
+          RETURNING ct.id AS thread_id
+        )
+        SELECT thread_id
+        FROM updated_threads;
```

Comment:
I believe, we don't need `WITH` here, the inner UPDATE query produces the same output, so it can be just:

```SQL
UPDATE chat_threads ct
SET replies_count = COALESCE(subquery.new_count, 0)
FROM (
  SELECT cm.thread_id, COUNT(cm.*) - 1 AS new_count
  FROM chat_threads
  LEFT JOIN chat_messages cm ON cm.thread_id = chat_threads.id AND cm.deleted_at IS NULL
  GROUP BY cm.thread_id
) AS subquery
WHERE ct.id = subquery.thread_id AND ct.replies_count IS DISTINCT FROM COALESCE(subquery.new_count, 0)
RETURNING ct.id AS thread_id
```

Reply:
Yes I think you are correct